### PR TITLE
wip(graph viz): initial bindings

### PR DESCRIPTION
**Pull request by lmeyerov**
Adds graphviz binding via pygraphviz

* Still requires out-of-band explicit install of `graphviz` + `graphviz-dev` native deps
* Supports using graphviz for layout and/or rendering
* Includes docs & demo [notebook](https://github.com/graphistry/pygraphistry/blob/23479a40f9795f32460c68734449ea9a62b39364/demos/demos_databases_apis/graphviz/graphviz.ipynb)

```python
g2 = g.layout_graphviz('dot')
g2.plot()
```

=>

Screenshot from 2024-10-03 03-18-11

## Security

* Safelists algs, attrs
* Toggle flag `drop_unsanitary=False` to `True` to reject graphviz attribs that may leak filesystem etc data
* Does not yet implement various `args: str` CLI params as pygraphviz does not sanitize them. We should allow when `drop_unsanitary=False`.

### CODE REVIEW

**Comment by lmeyerov** [comment]
Unsure why CI failing -- tests pass locally, TBD for serverside

File: demos/demos_databases_apis/graphviz/graphviz.ipynb [hunk 1]
```diff
@@ -0,0 +1,1143 @@
...
+        "* graphviz is generally not recommended for layout of graphs over 10,000 nodes and edges"
+      ],
+      "metadata": {
+        "id": "0BF6FBhDpLas"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#!sudo apt-get install graphviz graphviz-dev\n",
```
Comment (mj3cheun) [inline demos/demos_databases_apis/graphviz/graphviz.ipynb:51]: running sudo in a notebook doesn't really work. could either make it clear this needs to be run in a normal terminal or do something like https://stackoverflow.com/questions/44996933/using-sudo-inside-jupyter-notebooks-cell

File: demos/demos_databases_apis/graphviz/graphviz.ipynb [hunk 2]
```diff
@@ -0,0 +1,1143 @@
...
+      ],
+      "metadata": {
+        "id": "y7eC5hOCwfE1"
+      },
+      "execution_count": 33,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
```
Comment (mj3cheun) [inline demos/demos_databases_apis/graphviz/graphviz.ipynb:200]: what does this cell do?

File: demos/demos_databases_apis/graphviz/graphviz.ipynb [hunk 3]
```diff
@@ -0,0 +1,1143 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Static image rendering and entity-level attributes\n",
+        "\n",
+        "graphviz suports rendering to a static file in various image formats such as png.\n",
+        "\n",
+        "You can add special reserved-name columns to your node and edge dataframes to configure per-row render settings. These use the same names the above global attribute guidance."
```
Comment (mj3cheun) [inline demos/demos_databases_apis/graphviz/graphviz.ipynb:435]: typo in `These use the same names the above global attribute guidance.`

File: demos/demos_databases_apis/graphviz/graphviz.ipynb (hunk 1 above)
Comment (lmeyerov) [reply demos/demos_databases_apis/graphviz/graphviz.ipynb:51]: ah yeah tx will remove

File: demos/demos_databases_apis/graphviz/graphviz.ipynb (hunk 2 above)
Comment (lmeyerov) [reply demos/demos_databases_apis/graphviz/graphviz.ipynb:200]: deadcode tx

File: demos/demos_databases_apis/graphviz/graphviz.ipynb [hunk 4]
```diff
@@ -0,0 +1,1143 @@
...
+        "\n",
+        "\n",
+        "# Save a static graphviz render\n",
+        "g2c_positioned = g2c.layout_graphviz(\n",
+        "    \"dot\",\n",
+        "    render_to_disk=True,\n",
+        "    path=f'./graph.png',\n",
+        "    graph_attr={},\n",
+        "    edge_attr={},\n",
+        "    node_attr={'color': 'green'},\n",
```
Comment (mj3cheun) [inline demos/demos_databases_apis/graphviz/graphviz.ipynb:721]: this `node_attr` is overriden by color column assigned above

File: graphistry/plugins/graphviz.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    format: Optional[Format] = None,
+    drop_unsanitary: bool = False,
+) -> Plottable:
+    """
+
+    Use graphviz for layout, such as hierarchical trees and directed acycle graphs
+
+    Requires pygraphviz Python bindings and graphviz native libraries to be installed, see https://pygraphviz.github.io/documentation/stable/install.html
+
+    See PROGS for available layout algorithms
```
Comment (mj3cheun) [inline graphistry/plugins/graphviz.py:158]: i know that this is available in the source code but do we want to make `PROGS` and many others (e.g. GRAPH_ATTRS, NODE_ATTRS, etc.) available in the documentation or in the pygraphistry by exposing it?

Comment (lmeyerov) [reply graphistry/plugins/graphviz.py:158]: yes

**Comment by mj3cheun** [comment]
code looks good, just some docs stuff!

**Pull request merged**